equinor/vds-slice
# Abstract datahandle

This is a preliminary commit for the attribute difference(4D) endpoint.

The virtual differences cube is created by defining a virtual data source class used by all endpoints. Each new data source can now be created by implementing a new instance of the virtual class. This PR contains OvdsDataSource for accessing the *.vds files and OvdsMultiDataSource for data sources defined by a pair of *.vds files. Redefining the data source implies that current functions for attribute, fence and slice can be used without any changes. Thus, more code is reused and the the number of functions to tests is reduced. 

Remaining tasks:
- Test for new endpoints
- Test agains real data
- Load tests 
- Implement difference functions for fence and slice
- Verify output against real data
- ... Anything that is forgotten :-)
